### PR TITLE
Interpred tools test

### PR DESCRIPTION
**Pull request by Cidana-Developers**

### CODE REVIEW

File: test/ref/EbComputeMean_C.c [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ * ComputeMeanOfSquaredValues
+ *   returns the Mean of Squared Values
+ *******************************************/
+uint64_t ComputeMeanOfSquaredValues(
+    uint8_t* inputSamples,     // input parameter, input samples Ptr
+    uint32_t inputStride,      // input parameter, input stride
+    uint32_t inputAreaWidth,   // input parameter, input area width
+    uint32_t inputAreaHeight)  // input parameter, input area height
+{
+    uint32_t horizontalIndex;
```
Comment (andrey-norkin) [inline test/ref/EbComputeMean_C.c]: Variables in camelCase. Should be in snake_case.

File: test/ref/EbComputeMean_C.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                                      uint32_t input_area_width,
+                                      uint32_t input_area_height);
+
+/*******************************************
+ * ComputeMean
+ *   returns the mean of a block
+ *******************************************/
+uint64_t ComputeMean(
+    uint8_t* inputSamples,     // input parameter, input samples Ptr
+    uint32_t inputStride,      // input parameter, input stride
```
Comment (andrey-norkin) [inline test/ref/EbComputeMean_C.c]: variables in camel case.

Comment (Cidana-Developers) [reply test/ref/EbComputeMean_C.c]: Done.

File: test/ref/EbComputeMean_C.c (hunk 1 above)
Comment (Cidana-Developers) [reply test/ref/EbComputeMean_C.c]: Done.

**Comment by andrey-norkin** [comment]
There are some warnings in Unit Test build

**Comment by Cidana-Developers** [comment]
> There are some warnings in Unit Test build

Done, its clean now.

**Comment by andrey-norkin** [comment]
The Unit Test build on Linux passes without warnings now but the osx build still has some.

**Comment by Cidana-Developers** [comment]
> The Unit Test build on Linux passes without warnings now but the osx build still has some.

Rebased and removed the ld warnings of osx build and other warnings introduced recently.

**Comment by Cidana-Developers** [comment]
Regarding the failed cases (HmeTest and SadLoopSparseTest), they can be fixed by PR #444 and #458.

**Comment by mafonsoN** [comment]
@Cidana-Developers can you rebase and fix the conflicts?

**Comment by Cidana-Developers** [comment]
> @Cidana-Developers can you rebase and fix the conflicts?

Sure, it's done.

**Comment by mafonsoN** [comment]
@hassount any concerns about merging these tests?

File: CMakeLists.txt [hunk 3]
```diff
@@ -69,7 +69,6 @@ set(flags_to_test
     -Wformat
     -Wformat-security
     -fstack-protector-strong
-    -fPIE
```
Comment (hassount) [inline CMakeLists.txt]: why did we remove this cmake flag ?

**Comment by hassount** [comment]
> @hassount any concerns about merging these tests?

It's looks fine except for the compiler flag removed. it is needed as part of the requirements of our security team.

File: CMakeLists.txt (hunk 3 above)
Comment (Cidana-Developers) [reply CMakeLists.txt]: It's removed when we tried to eliminate the link warning on building shared libraries on osx:
`ld: warning: -pie being ignored. It is only used when linking a main executable`.
Will rebase it and add it back.

Comment (hassount) [reply CMakeLists.txt]: thank you

**Pull request merged**